### ISSUE TITLE
Flymake crashes with "Unexpected report from disabled backend"

### ISSUE THREAD

**Issue by m1cm1c** [orzechowskid/flymake-eslint#22]
This is an incomplete bug report as I don't know what's causing the problem.

As soon as I'm causing a linting violation in a file (and save that file), the following is printed to `*Messages*`:

```
error in process sentinel: flymake-error: [Flymake] Unexpected report from disabled backend flymake-eslint--checker
error in process sentinel: [Flymake] Unexpected report from disabled backend flymake-eslint--checker
```

Trying to go to the next error fails with the following message:
```
user-error: No more Flymake diagnostics
```

`flymake-eslint` is in my `package-selected-packages` in my init file. I'm using the following config:

```
(setq exec-path (append exec-path '("/path/to/project/node_modules/eslint/bin")))
(setq flymake-eslint-executable-name "/path/to/project/node_modules/eslint/bin/eslint.js")
(add-hook 'js-mode-hook (lambda ()
                          (flymake-eslint-enable)))
(add-hook 'js-mode-hook
	      (lambda()
	        (local-set-key (kbd "C-c C-f") 'flymake-goto-next-error)))
```

I'm using `eslint@8.13.0` and node version `v16.9.1` with GNU Emacs 27.2 on Manjaro. I'm not sure what additional information to provide. Please tell me.

**Comment by orzechowskid** [comment]
do you see any strange behavior if you lint the file manually?

**Comment by m1cm1c** [comment]
I can lint the entire project on the command line using

```
npx eslint --ext .js,.ts src
```

without any problems. Nothing is written to stderr.

The process exits with code 1 (nothing is printed about this; I queried it via `echo $?`) but I suspect that that's just because there are some open linting errors and warnings.

The only thing of note other than the above is that the output is prefixed with the following:

```
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.6.0

YOUR TYPESCRIPT VERSION: 4.6.2

Please only submit bug reports when using the officially supported version.

=============
```

I suspect that maybe this throws your code off because it causes the output not to start with a path to a file followed by a linting error or warning.

**Comment by m1cm1c** [comment]
The prefixed text about the typescript version cannot be the cause of the problem because the problem also occurs in a project that does not use typescript at all. It occurs in a file where this is the entire output before I made my breaking change (I did not open any text files prior to this in the same Emacs session; I only navigated to the file using dired):

```
$ npx eslint file.js 

/path/to/file.js
   41:1  warning  This line has a length of 124. Maximum allowed is 110  max-len
   42:1  warning  This line has a length of 114. Maximum allowed is 110  max-len
   43:1  warning  This line has a length of 118. Maximum allowed is 110  max-len
   47:1  warning  This line has a length of 116. Maximum allowed is 110  max-len
   51:1  warning  This line has a length of 117. Maximum allowed is 110  max-len
   53:1  warning  This line has a length of 117. Maximum allowed is 110  max-len
   57:1  warning  This line has a length of 119. Maximum allowed is 110  max-len
   59:1  warning  This line has a length of 122. Maximum allowed is 110  max-len
   63:1  warning  This line has a length of 130. Maximum allowed is 110  max-len
   76:1  warning  This line has a length of 112. Maximum allowed is 110  max-len
  102:1  warning  This line has a length of 122. Maximum allowed is 110  max-len
  103:1  warning  This line has a length of 137. Maximum allowed is 110  max-len
  108:1  warning  This line has a length of 136. Maximum allowed is 110  max-len
  114:1  warning  This line has a length of 125. Maximum allowed is 110  max-len
  123:1  warning  This line has a length of 212. Maximum allowed is 110  max-len
  146:1  warning  This line has a length of 168. Maximum allowed is 110  max-len
  148:1  warning  This line has a length of 124. Maximum allowed is 110  max-len
  149:1  warning  This line has a length of 125. Maximum allowed is 110  max-len
  152:1  warning  This line has a length of 118. Maximum allowed is 110  max-len
  154:1  warning  This line has a length of 148. Maximum allowed is 110  max-len
  156:1  warning  This line has a length of 136. Maximum allowed is 110  max-len
  158:1  warning  This line has a length of 136. Maximum allowed is 110  max-len
  163:1  warning  This line has a length of 120. Maximum allowed is 110  max-len
  173:1  warning  This line has a length of 120. Maximum allowed is 110  max-len
  189:1  warning  This line has a length of 172. Maximum allowed is 110  max-len
  192:1  warning  This line has a length of 129. Maximum allowed is 110  max-len
  194:1  warning  This line has a length of 140. Maximum allowed is 110  max-len
  197:1  warning  This line has a length of 124. Maximum allowed is 110  max-len
  199:1  warning  This line has a length of 130. Maximum allowed is 110  max-len
  215:1  warning  This line has a length of 130. Maximum allowed is 110  max-len
  216:1  warning  This line has a length of 138. Maximum allowed is 110  max-len
  218:1  warning  This line has a length of 114. Maximum allowed is 110  max-len
  219:1  warning  This line has a length of 161. Maximum allowed is 110  max-len
  221:1  warning  This line has a length of 475. Maximum allowed is 110  max-len
  242:1  warning  This line has a length of 164. Maximum allowed is 110  max-len
  243:1  warning  This line has a length of 131. Maximum allowed is 110  max-len
  247:1  warning  This line has a length of 116. Maximum allowed is 110  max-len
  249:1  warning  This line has a length of 136. Maximum allowed is 110  max-len
  263:1  warning  This line has a length of 168. Maximum allowed is 110  max-len
  265:1  warning  This line has a length of 114. Maximum allowed is 110  max-len
  268:1  warning  This line has a length of 162. Maximum allowed is 110  max-len
  269:1  warning  This line has a length of 130. Maximum allowed is 110  max-len
  272:1  warning  This line has a length of 153. Maximum allowed is 110  max-len
  281:1  warning  This line has a length of 167. Maximum allowed is 110  max-len
  282:1  warning  This line has a length of 158. Maximum allowed is 110  max-len
  285:1  warning  This line has a length of 167. Maximum allowed is 110  max-len
  286:1  warning  This line has a length of 158. Maximum allowed is 110  max-len
  295:1  warning  This line has a length of 167. Maximum allowed is 110  max-len
  296:1  warning  This line has a length of 158. Maximum allowed is 110  max-len
  299:1  warning  This line has a length of 195. Maximum allowed is 110  max-len
  300:1  warning  This line has a length of 176. Maximum allowed is 110  max-len
  301:1  warning  This line has a length of 195. Maximum allowed is 110  max-len
  302:1  warning  This line has a length of 172. Maximum allowed is 110  max-len
  327:1  warning  This line has a length of 168. Maximum allowed is 110  max-len
  328:1  warning  This line has a length of 133. Maximum allowed is 110  max-len
  329:1  warning  This line has a length of 168. Maximum allowed is 110  max-len
  330:1  warning  This line has a length of 133. Maximum allowed is 110  max-len
  333:1  warning  This line has a length of 130. Maximum allowed is 110  max-len
  334:1  warning  This line has a length of 141. Maximum allowed is 110  max-len
  335:1  warning  This line has a length of 130. Maximum allowed is 110  max-len
  336:1  warning  This line has a length of 141. Maximum allowed is 110  max-len
  339:1  warning  This line has a length of 126. Maximum allowed is 110  max-len
  340:1  warning  This line has a length of 131. Maximum allowed is 110  max-len
  341:1  warning  This line has a length of 126. Maximum allowed is 110  max-len
  342:1  warning  This line has a length of 131. Maximum allowed is 110  max-len
  345:1  warning  This line has a length of 167. Maximum allowed is 110  max-len
  346:1  warning  This line has a length of 133. Maximum allowed is 110  max-len
  347:1  warning  This line has a length of 167. Maximum allowed is 110  max-len
  348:1  warning  This line has a length of 133. Maximum allowed is 110  max-len
  351:1  warning  This line has a length of 207. Maximum allowed is 110  max-len
  352:1  warning  This line has a length of 136. Maximum allowed is 110  max-len
  369:1  warning  This line has a length of 140. Maximum allowed is 110  max-len
  370:1  warning  This line has a length of 136. Maximum allowed is 110  max-len
  373:1  warning  This line has a length of 140. Maximum allowed is 110  max-len
  374:1  warning  This line has a length of 147. Maximum allowed is 110  max-len
  377:1  warning  This line has a length of 160. Maximum allowed is 110  max-len
  378:1  warning  This line has a length of 147. Maximum allowed is 110  max-len
  381:1  warning  This line has a length of 160. Maximum allowed is 110  max-len
  382:1  warning  This line has a length of 170. Maximum allowed is 110  max-len
  389:1  warning  This line has a length of 141. Maximum allowed is 110  max-len
  390:1  warning  This line has a length of 147. Maximum allowed is 110  max-len
  393:1  warning  This line has a length of 161. Maximum allowed is 110  max-len
  394:1  warning  This line has a length of 147. Maximum allowed is 110  max-len
  397:1  warning  This line has a length of 161. Maximum allowed is 110  max-len
  398:1  warning  This line has a length of 170. Maximum allowed is 110  max-len
  408:1  warning  This line has a length of 134. Maximum allowed is 110  max-len
  418:1  warning  This line has a length of 217. Maximum allowed is 110  max-len
  425:1  warning  This line has a length of 140. Maximum allowed is 110  max-len
  437:1  warning  This line has a length of 173. Maximum allowed is 110  max-len
  438:1  warning  This line has a length of 140. Maximum allowed is 110  max-len
  444:1  warning  This line has a length of 197. Maximum allowed is 110  max-len
  447:1  warning  This line has a length of 149. Maximum allowed is 110  max-len
  448:1  warning  This line has a length of 134. Maximum allowed is 110  max-len
  452:1  warning  This line has a length of 183. Maximum allowed is 110  max-len
  453:1  warning  This line has a length of 197. Maximum allowed is 110  max-len
  454:1  warning  This line has a length of 153. Maximum allowed is 110  max-len
  457:1  warning  This line has a length of 149. Maximum allowed is 110  max-len
  461:1  warning  This line has a length of 141. Maximum allowed is 110  max-len
  464:1  warning  This line has a length of 157. Maximum allowed is 110  max-len
  465:1  warning  This line has a length of 145. Maximum allowed is 110  max-len
  480:1  warning  This line has a length of 187. Maximum allowed is 110  max-len
  481:1  warning  This line has a length of 154. Maximum allowed is 110  max-len
  484:1  warning  This line has a length of 188. Maximum allowed is 110  max-len
  485:1  warning  This line has a length of 155. Maximum allowed is 110  max-len
  490:1  warning  This line has a length of 118. Maximum allowed is 110  max-len
  494:1  warning  This line has a length of 183. Maximum allowed is 110  max-len
  497:1  warning  This line has a length of 135. Maximum allowed is 110  max-len
  498:1  warning  This line has a length of 130. Maximum allowed is 110  max-len
  503:1  warning  This line has a length of 118. Maximum allowed is 110  max-len
  507:1  warning  This line has a length of 117. Maximum allowed is 110  max-len
  511:1  warning  This line has a length of 117. Maximum allowed is 110  max-len
  513:1  warning  This line has a length of 114. Maximum allowed is 110  max-len
  515:1  warning  This line has a length of 200. Maximum allowed is 110  max-len
  516:1  warning  This line has a length of 193. Maximum allowed is 110  max-len
  517:1  warning  This line has a length of 130. Maximum allowed is 110  max-len
  521:1  warning  This line has a length of 111. Maximum allowed is 110  max-len
  527:1  warning  This line has a length of 134. Maximum allowed is 110  max-len
  538:1  warning  This line has a length of 195. Maximum allowed is 110  max-len
  539:1  warning  This line has a length of 168. Maximum allowed is 110  max-len
  548:1  warning  This line has a length of 134. Maximum allowed is 110  max-len
  558:1  warning  This line has a length of 217. Maximum allowed is 110  max-len
  559:1  warning  This line has a length of 217. Maximum allowed is 110  max-len
  566:1  warning  This line has a length of 140. Maximum allowed is 110  max-len
  578:1  warning  This line has a length of 173. Maximum allowed is 110  max-len
  579:1  warning  This line has a length of 140. Maximum allowed is 110  max-len
  585:1  warning  This line has a length of 197. Maximum allowed is 110  max-len
  588:1  warning  This line has a length of 149. Maximum allowed is 110  max-len
  589:1  warning  This line has a length of 134. Maximum allowed is 110  max-len
  593:1  warning  This line has a length of 149. Maximum allowed is 110  max-len
  597:1  warning  This line has a length of 184. Maximum allowed is 110  max-len
  598:1  warning  This line has a length of 151. Maximum allowed is 110  max-len
  600:1  warning  This line has a length of 181. Maximum allowed is 110  max-len
  601:1  warning  This line has a length of 151. Maximum allowed is 110  max-len
  605:1  warning  This line has a length of 141. Maximum allowed is 110  max-len
  620:1  warning  This line has a length of 115. Maximum allowed is 110  max-len
  630:1  warning  This line has a length of 163. Maximum allowed is 110  max-len
  631:1  warning  This line has a length of 147. Maximum allowed is 110  max-len
  649:1  warning  This line has a length of 116. Maximum allowed is 110  max-len
  656:1  warning  This line has a length of 147. Maximum allowed is 110  max-len
  657:1  warning  This line has a length of 147. Maximum allowed is 110  max-len
  659:1  warning  This line has a length of 164. Maximum allowed is 110  max-len
  661:1  warning  This line has a length of 175. Maximum allowed is 110  max-len
  663:1  warning  This line has a length of 151. Maximum allowed is 110  max-len
  666:1  warning  This line has a length of 147. Maximum allowed is 110  max-len
  667:1  warning  This line has a length of 151. Maximum allowed is 110  max-len
  670:1  warning  This line has a length of 148. Maximum allowed is 110  max-len
  671:1  warning  This line has a length of 151. Maximum allowed is 110  max-len
  674:1  warning  This line has a length of 148. Maximum allowed is 110  max-len
  675:1  warning  This line has a length of 151. Maximum allowed is 110  max-len
  677:1  warning  This line has a length of 143. Maximum allowed is 110  max-len
  678:1  warning  This line has a length of 117. Maximum allowed is 110  max-len
  680:1  warning  This line has a length of 116. Maximum allowed is 110  max-len
  681:1  warning  This line has a length of 132. Maximum allowed is 110  max-len
  682:1  warning  This line has a length of 140. Maximum allowed is 110  max-len
  684:1  warning  This line has a length of 147. Maximum allowed is 110  max-len
  685:1  warning  This line has a length of 145. Maximum allowed is 110  max-len
  686:1  warning  This line has a length of 163. Maximum allowed is 110  max-len
  687:1  warning  This line has a length of 174. Maximum allowed is 110  max-len
  689:1  warning  This line has a length of 150. Maximum allowed is 110  max-len
  692:1  warning  This line has a length of 145. Maximum allowed is 110  max-len
  693:1  warning  This line has a length of 150. Maximum allowed is 110  max-len
  696:1  warning  This line has a length of 146. Maximum allowed is 110  max-len
  697:1  warning  This line has a length of 150. Maximum allowed is 110  max-len
  700:1  warning  This line has a length of 146. Maximum allowed is 110  max-len
  701:1  warning  This line has a length of 150. Maximum allowed is 110  max-len
  704:1  warning  This line has a length of 115. Maximum allowed is 110  max-len
  707:1  warning  This line has a length of 149. Maximum allowed is 110  max-len
  708:1  warning  This line has a length of 151. Maximum allowed is 110  max-len
  710:1  warning  This line has a length of 116. Maximum allowed is 110  max-len
  712:1  warning  This line has a length of 144. Maximum allowed is 110  max-len
  713:1  warning  This line has a length of 118. Maximum allowed is 110  max-len
  715:1  warning  This line has a length of 142. Maximum allowed is 110  max-len
  716:1  warning  This line has a length of 116. Maximum allowed is 110  max-len
  722:1  warning  This line has a length of 122. Maximum allowed is 110  max-len
  723:1  warning  This line has a length of 142. Maximum allowed is 110  max-len
  725:1  warning  This line has a length of 136. Maximum allowed is 110  max-len
  728:1  warning  This line has a length of 117. Maximum allowed is 110  max-len
  731:1  warning  This line has a length of 222. Maximum allowed is 110  max-len
  732:1  warning  This line has a length of 141. Maximum allowed is 110  max-len
  735:1  warning  This line has a length of 164. Maximum allowed is 110  max-len
  738:1  warning  This line has a length of 222. Maximum allowed is 110  max-len
  739:1  warning  This line has a length of 141. Maximum allowed is 110  max-len
  741:1  warning  This line has a length of 222. Maximum allowed is 110  max-len
  742:1  warning  This line has a length of 141. Maximum allowed is 110  max-len
  744:1  warning  This line has a length of 222. Maximum allowed is 110  max-len
  745:1  warning  This line has a length of 141. Maximum allowed is 110  max-len
  747:1  warning  This line has a length of 222. Maximum allowed is 110  max-len
  748:1  warning  This line has a length of 141. Maximum allowed is 110  max-len
  750:1  warning  This line has a length of 222. Maximum allowed is 110  max-len
  751:1  warning  This line has a length of 141. Maximum allowed is 110  max-len
  753:1  warning  This line has a length of 222. Maximum allowed is 110  max-len
  754:1  warning  This line has a length of 141. Maximum allowed is 110  max-len
  756:1  warning  This line has a length of 222. Maximum allowed is 110  max-len
  757:1  warning  This line has a length of 141. Maximum allowed is 110  max-len
  759:1  warning  This line has a length of 222. Maximum allowed is 110  max-len
  760:1  warning  This line has a length of 141. Maximum allowed is 110  max-len
  785:1  warning  This line has a length of 177. Maximum allowed is 110  max-len
  791:1  warning  This line has a length of 166. Maximum allowed is 110  max-len
  798:1  warning  This line has a length of 224. Maximum allowed is 110  max-len
  799:1  warning  This line has a length of 174. Maximum allowed is 110  max-len
  802:1  warning  This line has a length of 186. Maximum allowed is 110  max-len
  805:1  warning  This line has a length of 151. Maximum allowed is 110  max-len
  806:1  warning  This line has a length of 147. Maximum allowed is 110  max-len
  817:1  warning  This line has a length of 140. Maximum allowed is 110  max-len
  825:1  warning  This line has a length of 198. Maximum allowed is 110  max-len
  826:1  warning  This line has a length of 138. Maximum allowed is 110  max-len
  829:1  warning  This line has a length of 173. Maximum allowed is 110  max-len
  832:1  warning  This line has a length of 125. Maximum allowed is 110  max-len
  833:1  warning  This line has a length of 134. Maximum allowed is 110  max-len

✖ 209 problems (0 errors, 209 warnings)
```

And this is the entire output after I made my breaking change (replaced a single-quoted string by a double-quoted string and entered a line break immediately after the opening quote):

```
$ npx eslint file.js 

/path/to/file.js
  21:37  error  Parsing error: Unterminated string constant

✖ 1 problem (1 error, 0 warnings)
```

**Comment by m1cm1c** [comment]
@orzechowskid Sorry, this problem is unrelated to your project. I simply didn't have `~/'.nvm/versions/node/v16.9.1/bin'` in my $PATH within Emacs because on my system, that extension of $PATH is was only performed for bash.

**Comment by m1cm1c** [comment]
To anyone else having this problem: It doesn't appear to be caused by the $PATH variable. But it's definitely something about the environment because the problem is gone if I start emacs via bash.
